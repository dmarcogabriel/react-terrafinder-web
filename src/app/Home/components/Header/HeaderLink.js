import React from 'react';
import { Link } from '@mui/material';

export const HeaderLink = ({ link, ...props }) => {
  return <Link sx={{ color: '#333' }} {...props} />;
};
