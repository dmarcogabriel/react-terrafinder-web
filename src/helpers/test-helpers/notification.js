export const mockUseNotification = (showNotification = () => {}) => {
  const mockShowNotification = jest.fn(showNotification);

  jest.mock('hooks/useNotification', () => ({
    useNotification: () => ({
      showNotification: mockShowNotification,
    }),
    NOTIFICATION_TYPES: { SUCCESS: 'success', ERROR: 'error' },
  }));

  return { mockShowNotification };
};
