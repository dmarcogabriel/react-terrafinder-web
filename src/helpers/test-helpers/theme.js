import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/themes';

export const withTheme = (component) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);

export const renderWithTheme = (component) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
