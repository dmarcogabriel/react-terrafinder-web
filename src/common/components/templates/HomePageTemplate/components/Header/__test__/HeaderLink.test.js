import React from 'react';
import ReactDOM from 'react-dom';
import { renderWithTheme, withTheme } from 'helpers/test-helpers/theme';
import { HeaderLink } from '../HeaderLink';

describe('', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(withTheme(<HeaderLink />), div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const tree = renderWithTheme(
      <HeaderLink to="test">Test Link</HeaderLink>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
