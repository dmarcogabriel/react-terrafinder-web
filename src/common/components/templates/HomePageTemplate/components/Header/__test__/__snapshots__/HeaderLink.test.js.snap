// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<a
  className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1rj4hyf-MuiTypography-root-MuiLink-root"
  onBlur={[Function]}
  onFocus={[Function]}
  to="test"
>
  Test Link
</a>
`;
